packages/app/src/cli/models/app/app.ts

```diff
@@ -343,6 +359,45 @@ export function validateFunctionExtensionsWithUiHandle(
   return errors.length > 0 ? errors : undefined
 }
 
+export type UIExtensionType = zod.infer<typeof UIExtensionSchema>
+
+export function validateExtensionsHandlesInCollection(
+  editorExtensionCollections: ExtensionInstance<EditorExtensionCollectionType>[],
+  allExtensions: ExtensionInstance[],
+): string[] | undefined {
+  const errors: string[] = []
+
+  const allowableTypesForExtensionInCollection = ['ui_extension']
+  editorExtensionCollections.forEach((collection) => {
+    const extensionsInCollection = makeExtensionsInCollection(collection.configuration)
+
+    extensionsInCollection.forEach((extension) => {
+      const matchingExtension = findExtensionByHandle(allExtensions, extension.handle)
+
+      if (!matchingExtension) {
+        errors.push(
+          `[${collection.handle}] editor extension collection - Local app must contain an extension with handle '${extension.handle}'`,
+        )
+      } else if (!allowableTypesForExtensionInCollection.includes(matchingExtension.specification.identifier)) {
+        errors.push(
+          `[${collection.handle}] editor extension collection - The collection can't contain an extension of type '${matchingExtension.specification.identifier}'`,
+        )
+      } else if (matchingExtension.specification.identifier === 'ui_extension') {
+        const uiExtension = matchingExtension as ExtensionInstance<UIExtensionType>
+        uiExtension.configuration.extension_points.forEach((extensionPoint) => {
+          if (extensionPoint.target.startsWith('admin.')) {
+            errors.push(
+              `[${collection.handle}] editor extension collection - The collection can't contain an extension of type '${matchingExtension.specification.identifier}' with target ${extensionPoint.target}`,
```

Comment:
cc @deannatron error message here

Reply:
New error copy:

>[${collection.handle}]: Remove extension of type 
'${matchingExtension.specification.identifier}' with target ${extensionPoint.target}` from this collection. This extension type is not supported in collections.

Reply:
Sorry Deanna. Here its not the type that is the issue, its just the target. Perhaps we can remove `This extension type is not supported in collections.` from this message?

Reply:
@oluwatimio actually that would be even more helpful. Like:

>[${collection.handle}]: Remove extension {extensionName} with target ${extensionPoint.target}` from this collection. This extension target is not supported in collections.

Ignore my syntax :D